packages/clay-time-picker/src/index.tsx

```diff
@@ -238,9 +238,33 @@ const ClayTimePicker: React.FunctionComponent<IProps> = ({
 				break;
 			default:
 				if (regex.test(event.key) && configName !== TimeType.ampm) {
+					const maxFirstDigit =
+						configName === TimeType.hours
+							? use12Hours
+								? 1
+								: 2
+							: 5;
+
+					const keyVal =
+						Number(event.key) < (config as ConfigMaxMin).min
+							? (config as ConfigMaxMin).min
+							: event.key;
+
+					const newVal =
+						Number(value) > maxFirstDigit
```

Comment:
You can use a variable that is already doing this, just so you don't have to convert again.

```suggestion
						intrinsicValue > maxFirstDigit
```